### PR TITLE
Fix: #182. Fix yaml support. Improve OAS3

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

Fix yaml support
Improves OAS3
Use snaphots for tests

### CODE REVIEW

File: datamodel_code_generator/parser/jsonschema.py [hunk 1]
```diff
@@ -633,10 +632,21 @@ def parse_raw_obj(self, name: str, raw: Dict[str, Any], path: List[str]) -> None
         self.parse_ref(obj, path)
 
     def parse_raw(self) -> None:
-        raw_obj: Dict[Any, Any] = json.loads(self.text)  # type: ignore
```
Comment (ioggstream) [inline datamodel_code_generator/parser/jsonschema.py:636]: Always use yaml parser so we don't care about file format.

File: datamodel_code_generator/parser/jsonschema.py [hunk 2]
```diff
@@ -633,10 +632,21 @@ def parse_raw_obj(self, name: str, raw: Dict[str, Any], path: List[str]) -> None
...
         obj_name = self.model_resolver.add([], obj_name, unique=False).name
         self.parse_raw_obj(obj_name, raw_obj, [])
-        definitions = raw_obj.get('definitions', {})
+        definitions, path = get_oas_schema(raw_obj)
         for key, model in definitions.items():
-            self.parse_raw_obj(key, model, ['#/definitions', key])
+            self.parse_raw_obj(key, model, [path, key])
+
+
+def get_oas_schema(oas_spec: Dict[Any, Any]):
```
Comment (ioggstream) [inline datamodel_code_generator/parser/jsonschema.py]: Support for different OAS schemas:
OAS2 - #/definitions
OAS3- #/components/schemas
...

File: datamodel_code_generator/parser/jsonschema.py (hunk 1 above)
Comment (koxudaxi) [reply datamodel_code_generator/parser/jsonschema.py:636]: Great :+1:

File: datamodel_code_generator/parser/jsonschema.py (hunk 2 above)
Comment (koxudaxi) [reply datamodel_code_generator/parser/jsonschema.py]: Sorry, I forgot the difference.
This file is for `JSON Schema`.
this method name should not have `OSA`.
Would you please remove `OSA` from the method ?

**Comment by koxudaxi** [comment]
@ioggstream 
Thank you for creating this PR.
I reviewed it.
Otherwise ok :smiley_cat:

File: datamodel_code_generator/parser/jsonschema.py (hunk 2 above)
Comment (ioggstream) [reply datamodel_code_generator/parser/jsonschema.py]: I just noticed you had another class for OAS3, and I moved out stuff there.
Should be fixed now.
We'll find a way to fix the usage of `definitions` in jsonschema another time :)

**Comment by ioggstream** [comment]
Should be ok. Take some time to review in the next days. :D

File: datamodel_code_generator/parser/jsonschema.py (hunk 2 above)
Comment (koxudaxi) [reply datamodel_code_generator/parser/jsonschema.py]: This code generator supported only OSA3 for the first time.
But, I supported JSON Schema by a request.

We will fix it next time:wink:

**Comment by koxudaxi** [comment]
I think this PR is fine.

I have merged this PR because I prefer a short iteration in this phase.

**Comment by koxudaxi** [comment]
@ioggstream 
I have released the version as `0.5.20`

Thank you very much.

**Pull request merged**